sdk/src/main/java/io/radar/sdk/RadarVerificationManager.kt

```diff
@@ -18,10 +29,322 @@ internal class RadarVerificationManager(
+
+                                                        callTrackApi(beacons)
+                                                    }
+                                                })
+                                        } else if (beacons != null) {
+                                            Radar.beaconManager.startMonitoringBeacons(beacons)
+
+                                            Radar.beaconManager.rangeBeacons(
+                                                beacons,
+                                                false,
+                                                object : Radar.RadarBeaconCallback {
+                                                    override fun onComplete(
+                                                        status: Radar.RadarStatus,
+                                                        beacons: Array<RadarBeacon>?
+                                                    ) {
+                                                        if (status != Radar.RadarStatus.SUCCESS || beacons == null) {
+                                                            callTrackApi(null)
+
+                                                            return
+                                                        }
+
+                                                        callTrackApi(beacons)
+                                                    }
+                                                })
+                                        } else {
+                                            callTrackApi(null)
+                                        }
+                                    }
+                                },
+                                false
+                            )
+                        } else {
+                            callTrackApi(null)
+                        }
+                    }
+                }
+            })
+    }
+
+    fun startTrackingVerified(token: Boolean, interval: Int, beacons: Boolean) {
+        val verificationManager = this
+
+        val request = NetworkRequest.Builder()
+            .addTransportType(NetworkCapabilities.TRANSPORT_ETHERNET)
+            .addTransportType(NetworkCapabilities.TRANSPORT_CELLULAR)
+            .addTransportType(NetworkCapabilities.TRANSPORT_BLUETOOTH)
+            .addTransportType(NetworkCapabilities.TRANSPORT_WIFI)
+            .removeCapability(NetworkCapabilities.NET_CAPABILITY_NOT_VPN)
+            .build()
+
+        connectivityManager.registerNetworkCallback(request, object : ConnectivityManager.NetworkCallback() {
+            override fun onAvailable(network: Network) {
+                super.onAvailable(network)
+
+                verificationManager.logger.d("Network connected")
+
+                if (token) {
+                    verificationManager.trackVerifiedToken(beacons)
+                } else {
+                    verificationManager.trackVerified(beacons)
+                }
```

Comment:
this runs immediately if network is available? Thinking through avoiding races between this and the interval below.

Maybe we should be setting or recommending some minimum interval as well given how long we know this to take?

Reply:
Shouldn't get called immediately, only on network change

The `handler.post` below will get called immediately. I also made sure a track happens immediately on iOS